src/hyperion/parameters/gridscan.py

```diff
@@ -0,0 +1,263 @@
+    def axis_1_start_um(self) -> float:
+        return self.axis_1.for_axis(self.x_start_um, self.y_start_um, self.z_start_um)
+
+    @property
+    def axis_2_start_um(self) -> float:
+        return self.axis_2.for_axis(self.x_start_um, self.y_start_um, self.z_start_um)
+
+    @property
+    def normal_axis_start(self) -> float:
+        return self.normal_axis.for_axis(
+            self.x_start_um, self.y_start_um, self.z_start_um
+        )
+
+    @property
+    def axis_1_end_um(self) -> float:
+        return self.axis_1_start_um + self.axis_1_step_size_um * self.axis_1_steps
+
+    @property
+    def axis_2_end_um(self) -> float:
+        return self.axis_2_start_um + self.axis_2_step_size_um * self.axis_2_steps
+
+    @property
+    def num_images(self) -> int:
+        return self.axis_1_steps * self.axis_2_steps
+
+    @property
+    def scan_spec(self):
+        line_1 = Line(
+            str(self.axis_1.value),
+            self.axis_1_start_um,
+            self.axis_1_end_um,
+            self.axis_1_steps,
+        )
+        line_2 = Line(
+            str(self.axis_2.value),
+            self.axis_2_start_um,
+            self.axis_2_end_um,
+            self.axis_2_steps,
+        )
+        return line_2 * ~line_1
+
+    @property
+    def scan_points(self):
+        return ScanPath(self.scan_spec.calculate()).consume().midpoints
+
+
+class ThreeDGridScan(SpecifiedGridScan):
+    demand_energy_ev: float | None = None
+    omega_start_deg: float | None = None
+    x_step_size_um: float = CONST.PARAM.GRIDSCAN.APERTURE_SIZE
+    y_step_size_um: float = CONST.PARAM.GRIDSCAN.APERTURE_SIZE
+    z_step_size_um: float = CONST.PARAM.GRIDSCAN.APERTURE_SIZE
+    y2_start_um: float
+    z2_start_um: float
+    x_steps: int
+    y_steps: int
+    z_steps: int
+
+    def _delete_excess(self, values: dict[str, Any]):
+        del values["x_step_size_um"]
+        del values["x_steps"]
+        del values["y2_start_um"]
+        del values["y_step_size_um"]
+        del values["y_steps"]
+        del values["z2_start_um"]
+        del values["z_step_size_um"]
+        del values["z_steps"]
+
+    @property
+    def scan_1(self) -> TwoDGridScan:
+        values = self.dict()
+        values["y_start_um"] = self.y_start_um
+        values["z_start_um"] = self.z_start_um
+        values["axis_1"] = XyzAxis.X
+        values["axis_2"] = XyzAxis.Y
+        values["axis_1_steps"] = self.x_steps
+        values["axis_2_steps"] = self.y_steps
+        self._delete_excess(values)
+        return TwoDGridScan(**values)
+
+    @property
+    def scan_2(self) -> TwoDGridScan:
+        values = self.dict()
+        values["y_start_um"] = self.y2_start_um
+        values["z_start_um"] = self.z2_start_um
+        values["axis_1"] = XyzAxis.X
+        values["axis_2"] = XyzAxis.Z
+        values["axis_1_steps"] = self.x_steps
+        values["axis_2_steps"] = self.z_steps
+        self._delete_excess(values)
+        return TwoDGridScan(**values)
```

Comment:
Should: This all feels like a messy way to represent the geometry of the system. Is it possible to leverage scanspec better to represent the geometry then use that to create the `GridScanParams`?

Reply:
Agreed that 3D and 2D grid scans are unrelated and don't need to refer to each other. Additionally, `scan_points` is the whole scan (including rotation in 3D case), `num_images` can just be derived from this but we then need a new property which is where to index the Nexus files to.